src/com/ms/silverking/io/StreamUtil.java

```diff
@@ -15,265 +13,264 @@
-            if (close && out != null) {
-                try {
-                    out.close();
-                } catch (IOException ioe_close) {
-                    ioe_close.printStackTrace();
-                }
-            }
+          out.close();
+        } catch (IOException ioe_close) {
+          ioe_close.printStackTrace();
         }
-        return numWritten;
-    }        
-
-    public static long streamToFile(InputStream in, File outFile, boolean close) throws IOException {
-        return stream(in, new FileOutputStream(outFile), close);
-    }
-    
-    public static long streamToFile(InputStream in, File outFile, boolean close, byte[] buf) throws IOException {
-        return stream(in, new FileOutputStream(outFile), close, buf);
-    }
-    
-    public static long streamToFile(InputStream in, String outFileName, boolean close) throws IOException {
-        return streamToFile(in, new File(outFileName), close);
+      }
     }
+    return numWritten;
+  }
 
-    public static long streamToFile(String s, File outFile, boolean close) throws IOException {
-        return streamToFile(stringToInputStream(s), outFile, close);
-    }
+  public static long stream(InputStream in, OutputStream out, boolean close, byte[] buf) throws IOException {
+    long numWritten;
 
-    public static long streamToFile(String s, File outFile) throws IOException {
-        return streamToFile(stringToInputStream(s), outFile, true);
-    }
-    
-    public static long streamToFile(String s, String outFileName, boolean close) throws IOException {
-        return streamToFile(s, new File(outFileName), close);
-    }
+    numWritten = 0;
+    try {
+      int numRead;
 
-    public static long streamToFile(String s, String outFileName) throws IOException {
-        return streamToFile(s, new File(outFileName), true);
-    }
-    
-    public static InputStream stringToInputStream(String s) {
-        return new ByteArrayInputStream( s.getBytes() );
+      numRead = 0;
+      do {
+        numRead = in.read(buf);
+        if (numRead > 0) {
+          out.write(buf, 0, numRead);
+          out.flush();
+          numWritten += numRead;
+        }
+      } while (numRead >= 0);
+    } finally {
+      out.flush();
+      if (close && in != null) {
+        try {
+          in.close();
+        } catch (IOException ioe_close) {
+          ioe_close.printStackTrace();
+        }
+      }
+      if (close && out != null) {
+        try {
+          out.close();
+        } catch (IOException ioe_close) {
+          ioe_close.printStackTrace();
+        }
+      }
     }
+    return numWritten;
+  }
+
+  public static long streamToFile(InputStream in, File outFile, boolean close) throws IOException {
+    return stream(in, new FileOutputStream(outFile), close);
+  }
+
+  public static long streamToFile(InputStream in, File outFile, boolean close, byte[] buf) throws IOException {
+    return stream(in, new FileOutputStream(outFile), close, buf);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileOutputStream*. The resource is referenced in statements at the following line: 250. The resource closure statement is at line: 236. There are other execution paths that do not contain closure statements, for example, when an else-branch is taken at line 234. Either a) close the object returned by *FileOutputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileOutputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).